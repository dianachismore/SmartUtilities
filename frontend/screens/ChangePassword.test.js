import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import ChangePassword from './ChangePassword';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

// Create a mock store
const mockStore = configureStore([]);

describe('ChangePassword Component', () => {
  let store;
  let component;
  let mockDispatch;

  beforeEach(() => {
    // Initialize the store with initial state if needed
    store = mockStore({});

    // Create a mock dispatch function
    mockDispatch = jest.fn();

    // Render the component with the mock store and dispatch function
    component = render(
      <Provider store={store}>
        <ChangePassword dispatch={mockDispatch} />
      </Provider>
    );
  });

  it('should render correctly', () => {
    expect(component).toMatchSnapshot();
  });

  it('should update oldPassword state when typing in the Old Password input', () => {
    const oldPasswordInput = component.getByPlaceholderText('Old Password');
    fireEvent.changeText(oldPasswordInput, 'oldPassword123');

    // Verify if the oldPassword state has been updated correctly
    expect(oldPasswordInput.props.value).toBe('oldPassword123');
  });

  it('should update newPassword state when typing in the New Password input', () => {
    const newPasswordInput = component.getByPlaceholderText('New Password');
    fireEvent.changeText(newPasswordInput, 'newPassword123');

    // Verify if the newPassword state has been updated correctly
    expect(newPasswordInput.props.value).toBe('newPassword123');
  });

});
