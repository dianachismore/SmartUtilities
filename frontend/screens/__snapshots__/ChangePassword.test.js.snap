// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChangePassword Component should render correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      {
        "fontSize": 20,
        "margin": 20,
      }
    }
  >
    Change Password
  </Text>
  <View
    style={
      {
        "width": "70%",
      }
    }
  >
    <TextInput
      onChangeText={[Function]}
      placeholder="Old Password"
      secureTextEntry={true}
      style={
        {
          "backgroundColor": "#fff",
          "borderColor": "#b5b5b5",
          "borderRadius": 15,
          "borderWidth": 1,
          "fontSize": 15,
          "marginVertical": 15,
          "padding": 10,
          "paddingLeft": 15,
        }
      }
      value=""
    />
    <TextInput
      onChangeText={[Function]}
      placeholder="New Password"
      secureTextEntry={true}
      style={
        {
          "backgroundColor": "#fff",
          "borderColor": "#b5b5b5",
          "borderRadius": 15,
          "borderWidth": 1,
          "fontSize": 15,
          "marginVertical": 15,
          "padding": 10,
          "paddingLeft": 15,
        }
      }
      value=""
    />
  </View>
  <View
    collapsable={false}
    style={
      {
        "backgroundColor": "#900",
        "borderColor": "transparent",
        "borderRadius": 4,
        "borderStyle": "solid",
        "borderWidth": 0,
        "elevation": 0,
        "minWidth": 64,
        "padding": 5,
        "shadowColor": "#000000",
        "shadowOffset": {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0,
        "shadowRadius": 0,
        "width": "70%",
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        {
          "disabled": false,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "overflow": "hidden",
          },
          {
            "borderRadius": 4,
          },
        ]
      }
    >
      <View
        style={
          [
            {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            },
            undefined,
          ]
        }
      >
        <Text
          numberOfLines={1}
          selectable={false}
          style={
            [
              {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              {
                "textAlign": "left",
              },
              [
                {
                  "letterSpacing": 1,
                  "marginHorizontal": 16,
                  "marginVertical": 9,
                  "textAlign": "center",
                },
                undefined,
                {
                  "textTransform": "uppercase",
                },
                {
                  "color": "#fff",
                  "fontFamily": "System",
                  "fontWeight": "500",
                },
                {
                  "fontFamily": "System",
                  "fontWeight": "500",
                },
                undefined,
              ],
            ]
          }
        >
          Change
        </Text>
      </View>
    </View>
  </View>
</View>
`;
